src/formula.jl

```diff
@@ -213,3 +213,23 @@ function Formula(t::Terms)
     append!(rhs.args,t.terms)
     Formula(lhs,rhs)
 end
+
+"""
+    dropterm(f::Formula, trm::Symbol)
+
+Return a copy of `f` without the term `trm`.
+
+# Examples
+```jl
+julia> dropterm(foo ~ 1 + bar + baz, :bar)
+foo ~ 1 + baz
+```
+"""
+function dropterm(f::Formula, trm::Symbol)
+    rhs = copy(f.rhs)
+    args = rhs.args
+    if !(Meta.isexpr(rhs, :call) && args[1] == :+ && (tpos = findlast(args, trm)) > 0)
```

Comment:
Yeah, that depends on the implicit intercept behavior. At the moment, I'd say it should return `y ~ 1` (e.g., the null model).

The `y ~ 0 + a` case is tricky, too; can you even evaluate `y ~ 0`?

I suppose you could just check whether the RHS would be empty and create `y ~ 1` for now, and we can fix it when we change intercepts to always be explicit.